### PR TITLE
Hypervisor: Fix IPv6StaticAddresses Patch operation

### PR DESCRIPTION
**Pull request by raviteja-b**
Currently VMI IPv6StaticAddresses patch operation fails due to a missing 1110 rebase commit.
These changes were missed as part of 1110 rebase PR  https://github.com/ibm-openbmc/bmcweb/pull/938

This commit fixes VMI IPv6StaticAddresses patch operation.

This commit requires the address and gateway. IPv4 network setting needs these values together to setup the IP. This is not a redfish need - but a networkd need. 

The basic VMI IP configuration is upstream, VMI is a IBM specific term so the more complex stuff is not upstream. https://github.com/ibm-openbmc/bmcweb/pull/938 would have to go upstream first. 

Tested by:
 PATCH -d '{"IPv6StaticAddresses": [{"Address": "2002:903:15f:290:9:3:109:39","PrefixLength": 120}]}'

### CODE REVIEW

File: redfish-core/lib/hypervisor_system.hpp [hunk 1]
```diff
@@ -938,32 +938,13 @@ inline void handleHypervisorIPv6StaticPatch(
...
-        {
-            if (!ip_util::ipv4VerifyIpAndGetBitcount(*address))
-            {
-                messages::propertyValueFormatError(asyncResp->res, *address,
-                                                   pathString + "/Address");
-                errorInEntry = true;
-            }
-        }
-        else
+        if (!address)
```
Comment (gtmills) [inline redfish-core/lib/hypervisor_system.hpp:941]: You have to patch with an address? 
Can't just do just the gateway, etc?

File: redfish-core/lib/hypervisor_system.hpp [hunk 2]
```diff
@@ -938,32 +938,13 @@ inline void handleHypervisorIPv6StaticPatch(
...
-        {
-            if (!ip_util::ipv4VerifyIpAndGetBitcount(*gateway))
-            {
-                messages::propertyValueFormatError(asyncResp->res, *gateway,
-                                                   pathString + "/Gateway");
-                errorInEntry = true;
-            }
-        }
-        else
+        if (!gateway)
```
Comment (gtmills) [inline redfish-core/lib/hypervisor_system.hpp:947]: Why do you have to patch with a gateway? 
Can't just do just the address, etc?

**Review by gtmills — COMMENTED** [review]
Upstream commit here? 
Can you explain more what this is fixing? 
This seems to add some logic requiring Gateway, Address, etc. Why is that needed?

File: redfish-core/lib/hypervisor_system.hpp [hunk 3]
```diff
@@ -976,13 +957,8 @@ inline void handleHypervisorIPv6StaticPatch(
...
         }
 
-        if (errorInEntry)
-        {
-            return;
-        }
-
-        BMCWEB_LOG_ERROR(
-            "INFO: Static ip configuration request from client: {} - IP Address: {}, Gateway: {}, Prefix Length: {}",
+        BMCWEB_LOG_DEBUG(
```
Comment (sunharis) [inline redfish-core/lib/hypervisor_system.hpp:960]: Please keep this as ERROR only. This was added for debugging dual HMC scenarios where we had hit some critical bugs at lab. Same comment for the other trace change as well.

Comment (raviteja-b) [reply redfish-core/lib/hypervisor_system.hpp]: done

File: redfish-core/lib/hypervisor_system.hpp (hunk 1 above)
Comment (raviteja-b) [reply redfish-core/lib/hypervisor_system.hpp:941]: This is downstream only code which was there in 1060 as well.
this was missed as part of rebase PRs.

File: redfish-core/lib/hypervisor_system.hpp (hunk 2 above)
Comment (raviteja-b) [reply redfish-core/lib/hypervisor_system.hpp:947]: This is downstream only code which was there in 1060 as well.
this was missed as part of rebase PRs.

**Comment by raviteja-b** [comment]
This PR fixes https://jazz07.rchland.ibm.com:13443/jazz/web/projects/CSSD#action=com.ibm.team.workitem.viewWorkItem&id=637213

File: redfish-core/lib/hypervisor_system.hpp (hunk 1 above)
Comment (asmithakarun) [reply redfish-core/lib/hypervisor_system.hpp:941]: >this was missed as part of rebase PRs.

+1

**Comment by gtmills** [comment]
Can you add more to the PR msg? 
Also having more in the commit msg helps. 
https://github.com/ibm-openbmc/bmcweb/pull/1035/commits/7ec56eeef37fa90c20a0e8aa08ec35a7032214a1 doesn't have a commit msg body. These make maintaining software easier.

File: redfish-core/lib/hypervisor_system.hpp [hunk 4]
```diff
@@ -976,13 +957,8 @@ inline void handleHypervisorIPv6StaticPatch(
...
         }
 
-        if (errorInEntry)
-        {
-            return;
-        }
-
         BMCWEB_LOG_ERROR(
-            "INFO: Static ip configuration request from client: {} - IP Address: {}, Gateway: {}, Prefix Length: {}",
+            "\nINFO: Static ip configuration request from client: {} - IP Address: {}, Gateway: {}, Prefix Length: {}",
```
Comment (gtmills) [inline redfish-core/lib/hypervisor_system.hpp:961]: Why the \n ?

**Review by sunharis — APPROVED** [review]

**Comment by raviteja-b** [comment]
> Can you add more to the PR msg? Also having more in the commit msg helps. [7ec56ee](https://github.com/ibm-openbmc/bmcweb/commit/7ec56eeef37fa90c20a0e8aa08ec35a7032214a1) doesn't have a commit msg body. These make maintaining software easier.

Done

**Comment by raviteja-b** [comment]
> Can you add more to the PR msg? Also having more in the commit msg helps. [7ec56ee](https://github.com/ibm-openbmc/bmcweb/commit/7ec56eeef37fa90c20a0e8aa08ec35a7032214a1) doesn't have a commit msg body. These make maintaining software easier.

done

**Review by asmithakarun — APPROVED** [review]
As this pulls in the missed changes from 1060 -> 1110 rebase, am good with this PR.

File: redfish-core/lib/hypervisor_system.hpp (hunk 2 above)
Comment (raviteja-b) [reply redfish-core/lib/hypervisor_system.hpp:947]: These changes were missed as part of 1110 rebase PR
https://github.com/ibm-openbmc/bmcweb/pull/938

**Review by gtmills — APPROVED** [review]

**Pull request merged**